packages/base/src/Internal/Static.hs

```diff
@@ -41,7 +44,14 @@ type ℝ = Double
 type ℂ = Complex Double
 
 newtype Dim (n :: Nat) t = Dim t
-  deriving Show
+  deriving (Show, Generic)
+
+instance Binary a => Binary (Complex a)
+  where
+    put (r :+ i) = put (r, i)
+    get = (\(r,i) -> r :+ i) <$> get
+
+instance (Binary a) => Binary (Dim n a)
```

Comment:
It would probably be a good idea to implement these instances manually, and check when parsing that the dimension is as specified